File: client/lib/core/models/message.dart
class Message {
  final int id;
  final String content;
  final int chatRoomId;
  final int userId;
  final String username;
  final String email;
  final DateTime timestamp;

  Message({
    required this.id,
    required this.content,
    required this.chatRoomId,
    required this.userId,
    required this.username,
    required this.email,
    required this.timestamp,
  });

  factory Message.fromJson(Map<String, dynamic> json) {
    return Message(
      id: json['id'],
      content: json['content'],
      chatRoomId: json['chat_room_id'],
      userId: json['user_id'],
      username: json['username'],
      email: json['email'],
      timestamp: DateTime.parse(json['timestamp']),
    );
  }
}

File: backend/internal/models/Message.go
package models

import "time"

// Message représente un message dans une salle de chat
type Message struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	ChatRoomID uint      `json:"chat_room_id"`
	UserID     uint      `json:"user_id"`
	Content    string    `json:"content"`
	Timestamp  time.Time `json:"timestamp"`
}

type MessageAdd struct {
	Email   string `json:"email"`
	Content string `json:"content"`
}

type MessageResponse struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	ChatRoomID uint      `json:"chat_room_id"`
	UserID     uint      `json:"user_id"`
	Email      string    `json:"email" required:""`
	Username   string    `json:"username" required:""`
	Content    string    `json:"content"`
	Timestamp  time.Time `json:"timestamp"`
}
